crates/core/app/tests/mock_consensus_staking.rs

```diff
@@ -101,29 +101,23 @@ async fn mock_consensus_can_delegate_to_a_validator() -> anyhow::Result<()> {
 
     // Make a transaction that defines a new validator.
     let new_validator_vk = node.keyring_mut().add_key();
-    let (new_validator_pb, _) = self::common::generate_penumbra_validator(&new_validator_vk);
+    let (new_validator_pb, _, new_validator_spend_key) =
+        self::common::generate_penumbra_validator(&new_validator_vk);
```

Comment:
This structure has inverted control flow to what we want. We don't want to be using the engine to generate the keys. We want to be defining a validator and passing it to the engine.

Reply:
In particular, we **don't** want to be using a common helper method to define the new validator we're adding. It should be possible to reason about the semantics of the test function just from reading its code, but if we reuse an internal helper method of the `auto_app_state` implementation, we're coupling the semantics of the test to the implementation of some other part of the test harness. This is actively undesirable.

Reply:
example here: ceb5d6283

Reply:
> we don't want to be using a common helper method to define the new validator we're adding. It should be possible to reason about the semantics of the test function just from reading its code, but if we reuse an internal helper method of the auto_app_state implementation, we're coupling the semantics of the test to the implementation of some other part of the test harness. This is actively undesirable.

definitely. i agree. it was a mistake to reuse that helper function here. i have cherry-picked ceb5d62 into this branch.